diff --git a/cmake/onnxruntime.cmake b/cmake/onnxruntime.cmake
index 927b4ac..85afaa7 100644
--- a/cmake/onnxruntime.cmake
+++ b/cmake/onnxruntime.cmake
@@ -249,7 +249,8 @@ if(NOT APPLE AND NOT WIN32)
       LINK_DEPENDS ${SYMBOL_FILE}
       VERSION ${ORT_VERSION}
       SOVERSION 1
-      FOLDER "ONNXRuntime")
+      FOLDER "ONNXRuntime"
+      PROPERTIES NO_SONAME 1) # AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
   endif()
 else()
   # Omit the SOVERSION setting in Windows/macOS/iOS/.. build
